### PR TITLE
Add `@Flaky` annotation to Databricks product tests

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Add `@Flaky` annotation to Databricks product tests
Relates to #14391

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Comment by findinpath** [comment]
I've found that the following tests are missing the annotation:

- io.trino.tests.product.deltalake.TestDeltaLakeAlterTableCompatibility#testTrinoAlterTablePreservesTableMetadata
     (probably not needed because it performs only a CREATE TABLE statement on delta )
- io.trino.tests.product.deltalake.TestDeltaLakeAlterTableCompatibility#testTrinoAlterTablePreservesTableMetadata 
     (probably not needed because it performs only a CREATE TABLE statement on delta )
- io.trino.tests.product.deltalake.TestDeltaLakeAlterTableCompatibility#testTrinoAlterTablePreservesGeneratedColumn
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksCheckpointsCompatibility#testTrinoCheckpointNullStatisticsForRowType
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksCheckpointsCompatibility#testDatabricksCheckpointNullStatisticsForRowType
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksCheckpointsCompatibility#testTrinoWriteStatsAsJsonDisabled
     (probably not needed)
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksCheckpointsCompatibility#testDatabricksWriteStatsAsJsonDisabled
     (probably not needed)
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksCheckpointsCompatibility#testTrinoWriteStatsAsStructDisabled
     (probably not needed)
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksCheckpointsCompatibility#testDatabricksWriteStatsAsStructDisabled
      (probably not needed)
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksInsertCompatibility#testCheckConstraintsCompatibility
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksInsertCompatibility#testCompression(java.lang.String)
     (probably not needed)
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksInsertCompatibility#testWritesToTableWithCheckConstraintFails
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksInsertCompatibility#testMetadataOperationsRetainCheckConstraints
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksInsertCompatibility#testWritesToTableWithGeneratedColumnFails
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksInsertCompatibility#testWritesToTableWithCDFFails
- io.trino.tests.product.deltalake.TestHiveAndDeltaLakeRedirect#testInsertIntoDeltaTableFromHiveNonDefaultSchemaRedirect
- io.trino.tests.product.deltalake.TestHiveAndDeltaLakeRedirect#testViewReferencingHiveAndDeltaTable

**Comment by ebyhr** [comment]
Not sure why IntelliJ returned wrong results. Anyway, added the annotation to all of `DELTA_LAKE_DATABRICKS` regardless of the detailed tests.

**Comment by findinpath** [comment]
You're still missing 
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksCheckpointsCompatibility#testTrinoCheckpointNullStatisticsForRowType
- io.trino.tests.product.deltalake.TestDeltaLakeDatabricksCheckpointsCompatibility#testDatabricksCheckpointNullStatisticsForRowType

**Review by findinpath — APPROVED** [review]

**Comment by ebyhr** [comment]
Pushed an empty commit because there was no retrying with "Communication link failure".

**Comment by ebyhr** [comment]
https://github.com/trinodb/trino/actions/runs/3407215065/jobs/5666758475 has retrying with "Communication link failure".

**Review by findinpath — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/util/DeltaLakeTestUtils.java [hunk 1]
```diff
@@ -25,6 +26,11 @@
     public static final String DATABRICKS_104_RUNTIME_VERSION = "10.4";
     public static final String DATABRICKS_91_RUNTIME_VERSION = "9.1";
 
+    public static final String DATABRICKS_COMMUNICATION_FAILURE_ISSUE = "https://github.com/trinodb/trino/issues/14391";
```
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/util/DeltaLakeTestUtils.java:29]: https://github.com/trinodb/trino/issues/14391

**Review by findepi — APPROVED** [review]

**Pull request merged**